### PR TITLE
Update PORTA to v1.4.2

### PR DESCRIPTION
**Pull request by bdoolittle**
Updating the build script for the PORTA_jll

### CODE REVIEW

**Comment by bdoolittle** [comment]
This branch is ready for merge from my end.

File: P/PORTA/build_tarballs.jl [hunk 1]
```diff
@@ -3,29 +3,30 @@
...
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release julia-porta
+make -j
+make install
+install_license julia-porta/LICENSE
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = supported_platforms()
+platforms = supported_platforms(experimental=false)
```
Comment (giordano) [inline P/PORTA/build_tarballs.jl:24]: This does nothing
```suggestion
platforms = supported_platforms()
```

File: P/PORTA/build_tarballs.jl [hunk 2]
```diff
@@ -3,29 +3,30 @@
...
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir
-mkdir -p ${bindir}
-make -C julia-porta/gnu-make/ CC=${CC}
-cp julia-porta/gnu-make/bin/xporta${exeext} ${bindir}
-cp julia-porta/gnu-make/bin/valid${exeext} ${bindir}
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release julia-porta
+make -j
```
Comment (giordano) [inline P/PORTA/build_tarballs.jl:17]:
```suggestion
make -j{nproc}
```

File: P/PORTA/build_tarballs.jl [hunk 3]
```diff
@@ -3,29 +3,30 @@
...
 
 # Collection of sources required to complete build
 sources = [
-    GitSource("https://github.com/bdoolittle/julia-porta.git", "93b99656f390c2252a0142bf9a4c22f5217a42da")
+    GitSource("https://github.com/bdoolittle/julia-porta.git", "41a7653c0df6dc0fe738aa4a2347096f27d391c0")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir
```
Comment (giordano) [inline P/PORTA/build_tarballs.jl:15]:
```suggestion
cd $WORKSPACE/srcdir/julia-porta
```

File: P/PORTA/build_tarballs.jl [hunk 4]
```diff
@@ -3,29 +3,30 @@
...
-mkdir -p ${bindir}
-make -C julia-porta/gnu-make/ CC=${CC}
-cp julia-porta/gnu-make/bin/xporta${exeext} ${bindir}
-cp julia-porta/gnu-make/bin/valid${exeext} ${bindir}
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release julia-porta
```
Comment (giordano) [inline P/PORTA/build_tarballs.jl:16]:
```suggestion
mkdir build && cd build
cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release ..
```

File: P/PORTA/build_tarballs.jl [hunk 5]
```diff
@@ -3,29 +3,30 @@
...
+make -j
+make install
+install_license julia-porta/LICENSE
```
Comment (giordano) [inline P/PORTA/build_tarballs.jl:19]:
```suggestion
install_license ../LICENSE
```

**Comment by bdoolittle** [comment]
Thanks for the feedback @giordano, makes sense to me

File: P/PORTA/build_tarballs.jl [hunk 6]
```diff
@@ -3,20 +3,21 @@
...
 script = raw"""
-cd $WORKSPACE/srcdir
-mkdir -p ${bindir}
-make -C julia-porta/gnu-make/ CC=${CC}
-cp julia-porta/gnu-make/bin/xporta${exeext} ${bindir}
-cp julia-porta/gnu-make/bin/valid${exeext} ${bindir}
+cd $WORKSPACE/srcdir/julia-porta
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release ..
+make -j{nproc}
```
Comment (giordano) [inline P/PORTA/build_tarballs.jl:18]: Ooops, my fault
```suggestion
make -j${nproc}
```

**Pull request merged**